### PR TITLE
 Reviewed tau storage in NanoAOD

### PR DESCRIPTION
**Pull request by kandrosov**
- Using RecoTauTag.RecoTau.tools.runTauIdMVA to rerun taus ids.
- Removed taus_updatedMVAIds_cff.py
- Removed old ids, which are not stored in nanoAOD, from the tau preselection.
- Added deepTau2017v2 variables.
- Removed pre-102X eras.

### CODE REVIEW

File: PhysicsTools/NanoAOD/python/taus_cff.py [hunk 1]
```diff
@@ -1,43 +1,38 @@
...
-from PhysicsTools.NanoAOD.taus_updatedMVAIds_cff import *
+from PhysicsTools.JetMCAlgos.TauGenJetsDecayModeSelectorAllHadrons_cfi import tauGenJetsSelectorAllHadrons
 
 ##################### User floats producers, selectors ##########################
 
-
 finalTaus = cms.EDFilter("PATTauRefSelector",
-    src = cms.InputTag("slimmedTausUpdated"),
-    cut = cms.string("pt > 18 && tauID('decayModeFindingNewDMs') && (tauID('byLooseCombinedIsolationDeltaBetaCorr3Hits') || tauID('byVLooseIsolationMVArun2v1DBoldDMwLT2015') || tauID('byVLooseIsolationMVArun2v1DBnewDMwLT') || tauID('byVLooseIsolationMVArun2v1DBdR03oldDMwLT') || tauID('byVVLooseIsolationMVArun2v1DBoldDMwLT') || tauID('byVVLooseIsolationMVArun2v1DBoldDMwLT2017v2') || tauID('byVVLooseIsolationMVArun2v1DBnewDMwLT2017v2') || tauID('byVVLooseIsolationMVArun2v1DBdR03oldDMwLT2017v2'))")
+    src = cms.InputTag("slimmedTausNewID"),
```
Comment (mbluj) [inline PhysicsTools/NanoAOD/python/taus_cff.py]: Propose to keep previously used name of updated tau collection, i.e. `slimmedTausUpdated` (it applies also to other places in `PhysicsTools/NanoAOD/python/nano_cff.py`)

File: PhysicsTools/NanoAOD/python/nano_cff.py [hunk 2]
```diff
@@ -145,6 +145,16 @@
     modifier.toModify(extraFlagsTable, variables= cms.PSet())
     modifier.toModify(extraFlagsTable, variables = dict(Flag_ecalBadCalibFilterV2 = ExtVar(cms.InputTag("ecalBadCalibFilterNanoTagger"), bool, doc = "Bad ECAL calib flag (updated xtal list)")))
 
+import RecoTauTag.RecoTau.tools.runTauIdMVA as tauIdConfig
```
Comment (mbluj) [inline PhysicsTools/NanoAOD/python/nano_cff.py:149]: Add comment about goal of the new function, something like "Tool to add new TauIDs" or so. 
I propose to keep previously used name of updated tau collection, i.e. `slimmedTausUpdated` (check also other places).

**Review by mbluj — CHANGES_REQUESTED** [review]
Hello,
first at all thank you for implementing this. 
I think the implementation is fine to provide it to users, but not ideal to proceed to official CMSSW/NanoAOD due to missing/removed support of eras. 
Please include also a small comment put inline. In meantime, I will run it with some data for a functional test.

**Comment by kandrosov** [comment]
> I think the implementation is fine to provide it to users, but not ideal to proceed to official CMSSW/NanoAOD due to missing/removed support of eras.

Support of eras can be added back, but I would need some guidance to do it. On the other hand, I don't think that there will be any new central nanoAOD production with 94X, 92X or 80X. Are we sure that the back compatibility with those eras is still necessary?

**Comment by mbluj** [comment]
Let's forget on eras now in this PR to branch not expected for integration with official CMSSW.
Just for records: 
* people want to keep eras as the era are used also for testing purposes,
* name of CMSSW version in name of era in many cases reflects version of inputs. In many cases it is possible to produce consistent NanoAOD using MiniAOD produced with various versions of CMSSW as part of various campaigns.

**Comment by mbluj** [comment]
Tests are successful: 
* new implementation provides DeepTauIDs to NanoAOD outputs,
* distributions of MVA-based TauIDs are identical to those with unmodified (previous) configuration (with events passing same preselection).

=> merging

@kandrosov please update the DeepTau 2017v2 recipe at TWiki (add instruction on NanoAOD) when this PR is merged. Then you can advertise this via HNs and ask conveners to add this to their news slides.

**Pull request merged**